discordbot/bot/dictionary-module/mwdict.go

```diff
@@ -11,9 +11,9 @@ import (
 type Client mwdict.Client
 
 // NewClient creates a new client for the dictionary module
-func NewClient(apiKey config.MerriamWebsterAPIKey, userAgent config.UserAgent) (*mwdict.BasicClient, error) {
+func NewClient(apiKey config.MerriamWebsterAPIKey, userAgent config.UserAgent) *mwdict.BasicClient {
 	if apiKey == "" {
-		return nil, errors.New("no Merriam-Webster API key found")
+		return mwdict.NewBasicClient("", string(userAgent), errors.New("no Merriam-Webster API key found"))
```

Comment:
This line is actually unnecessary. `config.MerriamWebsterAPIKey` is an alias for `string` already, so the single line using it as `string(apiKey)` is enough to turn it back to the `""` it came in as.

Reply:
Also, as commented below, this logic makes more sense in NewBasicClient.

Come to think of it, this is a few too many layers in the first place. Might need to refactor this to not have so much redundant crap. Still, put it in the other place, please :).